packages/x-date-pickers/src/models/fields.ts

```diff
@@ -11,6 +11,8 @@ export type FieldSectionType =
   | 'seconds'
   | 'meridiem';
 
+export type FieldSectionContentType = 'digit' | 'digit-with-letter' | 'letter';
```

Comment:
I'm creating a third content type `digit-with-letter` because the behavior starts to diverge to much from `digit` and I needed to generate dates in a lot of places just to check if there was letter in the result.
Having it hard-coded in the section makes it much easier to reason with.